lib/runtime/wasmer/instance.go

```diff
@@ -156,25 +156,24 @@ func (in *Instance) UpdateRuntimeCode(code []byte) (err error) {
 	return nil
 }
 
-// CheckRuntimeVersion calculates runtime Version for runtime blob passed in
-func (in *Instance) CheckRuntimeVersion(code []byte) (runtime.Version, error) {
-	in.mutex.Lock()
-	defer in.mutex.Unlock()
-
-	wasmInstance, allocator, err := setupVM(code)
+// CheckRuntimeVersion finds the runtime version by initiating a temporary
+// runtime instance using the WASM code provided, and querying it.
+func CheckRuntimeVersion(code []byte) (version runtime.Version, err error) {
```

Comment:
```suggestion
func GetRuntimeVersion(code []byte) (version runtime.Version, err error) {
```
even though I don't like `GetXXX` syntax in go, I think it's better than `Check..` since it's not checking anything.

Reply:
Or `QueryRuntimeVersion`?

Reply:
Actually I'm setting it as `GetRuntimeVersion` since a bunch of other methods are also called `GetRuntimeVersion`, for consistency. Maybe we can rename all of them to `QueryRuntimeVersion` later.